proposals/0000-markdown-content-redesign.md

```diff
@@ -0,0 +1,126 @@
+- Start Date: 03-10-2022
+- Reference Issues: https://github.com/withastro/rfcs/discussions/5, https://github.com/withastro/rfcs/discussions/118
+- Implementation PR: <!-- leave empty -->
+
+# Summary
+
+A v2.0 API for working with local Markdown files in Astro.
+
+# Example
+
+```astro
+---
+// BASIC USAGE:
+
+// Note: Astro.fetchContent() removed in favor of direct `import.meta.globEager` usage
+// Note: This works in non-Astro framework components!
+const markdownFilesObj = import.meta.globEager('../posts/*.md');
+const markdownFilesArr = Object.values(markdownFilesObj);
+const markdownFilesFiltered = markdownFilesArr.filter(post => post.data.category === 'blog');
+
+// Note: Individual markdown imports are now supported as well.
+const firstPost = await import('../posts/first-post.md');
```

Comment:
This is a very cool feature, it's nice being able to just use `await import()` here rather than using an API designed for loading multiple files with a glob pattern